bot/cogs/emote.py

```diff
@@ -0,0 +1,167 @@
+from enum import Enum
+
+import aiohttp
+import disnake
+from disnake.ext import commands
+from disnake.i18n import Localized
+
+from bot.core import Bot
+from bot.services import pil
+
+# Source: https://discord.com/blog/beginners-guide-to-custom-emojis#heading-4
+MAX_IMAGE_BYTES_SIZE = 256 * 1000
+NO_PERM_GIF = "https://media1.tenor.com/m/8wwLg1lZS_gAAAAC/denied-incredibles.gif"
+LARGE_GIF = "https://media1.tenor.com/m/SGCLBFPTK4sAAAAC/im-big-boned-eric-cartman.gif"
+CHOOSE_GIF = "https://media1.tenor.com/m/3C4bAf7b8C0AAAAC/the-matrix-morpheus.gif"
+
+
+class EmoteEmbed(disnake.Embed):
+    def __init__(
+        self,
+        *,
+        name: str,
+        image: disnake.File,
+        inter: disnake.GuildCommandInteraction,
+    ) -> None:
+        locale = inter.locale.name.replace("_", "-")
+        title = inter.bot.i18n.get(key="COMMAND_EMOTE_ADD_TITLE")[locale]
+        title = title.format(name)
+
+        super().__init__(
+            title=title,
+            color=disnake.Colour.green(),
+        )
+        self.set_image(file=image)
+
+
+class EmoteErrorEnum(Enum):
+    WITHOUT_PERMISSION = 1
+    LARGE_IMAGE = 2
+    NO_EMOTE_ARGS = 3
+
+
+class EmoteError(disnake.Embed):
+    def __init__(
+        self,
+        *,
+        name: str,
+        err_enum: EmoteErrorEnum | None = None,
+        inter: disnake.GuildCommandInteraction,
+    ) -> None:
+        locale = inter.locale.name.replace("_", "-")
+        gif_error = None
+        description = "IDK"
+
+        title = inter.bot.i18n.get(key="COMMAND_EMOTE_ADD_ERR_TITLE")[locale]
+        title = title.format(name)
+
+        match err_enum:
+            case EmoteErrorEnum.WITHOUT_PERMISSION:
+                gif_error = NO_PERM_GIF
+                description = inter.bot.i18n.get(key="COMMAND_EMOTE_ADD_ERR_WITHOUT_PERM")[locale]
+
+            case EmoteErrorEnum.LARGE_IMAGE:
+                gif_error = LARGE_GIF
+                description = inter.bot.i18n.get(key="COMMAND_EMOTE_ADD_ERR_LARGE_IMAGE")[locale].format(
+                    "https://discord.com/blog/beginners-guide-to-custom-emojis#heading-4"
+                )
+
+            case EmoteErrorEnum.NO_EMOTE_ARGS:
+                gif_error = CHOOSE_GIF
+                description = "USE UM DOS DOIS ARGUMENTOS OPCIONAIS"
+
+        super().__init__(
+            title=title,
+            description=description,
+            color=disnake.Colour.red(),
+        )
+
+        self.set_image(url=gif_error)
+
+
+class Emote(commands.Cog):
+    def __init__(self, bot: Bot) -> None:
+        self.bot = bot
```

Comment:
Não tem necessidade desse construtor se não for utilizar o `self.bot`, pode remover o trecho:
```py
def __init__(self, bot: Bot) -> None:
    self.bot = bot
```